signer/src/config/mod.rs

```diff
@@ -130,9 +136,48 @@ pub struct BitcoinConfig {
     #[serde(deserialize_with = "url_deserializer_vec")]
     pub rpc_endpoints: Vec<Url>,
 
-    /// Bitcoin ZeroMQ block-hash stream endpoint.
-    #[serde(deserialize_with = "url_deserializer_vec")]
-    pub block_hash_stream_endpoints: Vec<Url>,
+    /// The number of seconds to wait between polling for new canonical block
+    /// hashes (`getbestblockhash`).
+    #[serde(deserialize_with = "duration_seconds_deserializer")]
+    pub chain_tip_polling_interval: std::time::Duration,
+}
+
+impl Validatable for BitcoinConfig {
+    fn validate(&self, _: &Settings) -> Result<(), ConfigError> {
+        // At least one endpoint must be provided.
+        if self.rpc_endpoints.is_empty() {
+            return Err(ConfigError::Message(
+                "[bitcoin.rpc_endpoints] At least one Bitcoin RPC endpoint must be provided"
+                    .to_string(),
+            ));
+        }
+
+        // Validate each endpoint configuration.
+        for endpoint in &self.rpc_endpoints {
+            if !["http", "https"].contains(&endpoint.scheme()) {
+                return Err(ConfigError::Message(
+                    "[bitcoin.rpc_endpoints] Invalid URL scheme: must be HTTP or HTTPS".to_string(),
+                ));
+            }
+
+            if endpoint.host_str().is_none() {
+                return Err(ConfigError::Message(
+                    "[bitcoin.rpc_endpoints] Invalid URL: host is required".to_string(),
+                ));
+            }
+        }
+
+        if self.chain_tip_polling_interval.as_secs()
+            > MAX_BITCOIN_CHAIN_TIP_POLLING_INTERVAL_SECONDS
```

Comment:
nit: can we add a test for this?

Reply:
Good thing you asked, because I had forgotten to call `validate()` 🙈 [4a56f88](https://github.com/stacks-sbtc/sbtc/pull/1780/commits/4a56f88dbaa60eef0f6e6740824abd2e8e357f97)